### PR TITLE
Add promise support and tests

### PR DESCRIPTION
**Pull request by grantholle**
There's support to use es6 native promises in lieu of callbacks. Callbacks are still supported, though. I also added tests that covers most of the features (although not all) using both callbacks and promises. I also discovered a few functions didn't work right (I think mostly the delete functions), and those are now working properly as well.

### CODE REVIEW

**Comment by alexwhitman** [comment]
Thanks for this.  I haven't had time to look at it properly yet but I will get around to it soon.

**Comment by grantholle** [comment]
Have you had a chance to check it out yet?

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules/
+.vscode
```
Comment (alexwhitman) [inline .gitignore]: This should be in a global `.gitignore` really (https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).  I prefer to keep only repository specific files in the respository `.gitignore`.

File: lib/internal/pushes.js [hunk 2]
```diff
@@ -45,75 +46,71 @@ PushBullet.prototype.link = function link(deviceParams, title, url, callback) {
  * @param  {String}   filePath     Path to file.
  * @param  {String}   message      Optional message to send with file.
  * @param  {Function} callback     Callback for when the request is complete.
+ * @returns {Promise}
  */
-PushBullet.prototype.file = function file(deviceParams, filePath, message, callback) {
```
Comment (alexwhitman) [inline lib/internal/pushes.js]: A message/body is a valid parameter for file and shouldn't be removed. https://docs.pushbullet.com/#parameters-for-different-types-of-pushes

**Comment by grantholle** [comment]
Made the requested changes, and also fixed #31 while I was in there which provides support for the body parameter on the link method.

**Comment by alexwhitman** [comment]
Sorry, something else I've just noticed.  You've used ES6 default function parameters in a few places.  These are only supported from 6.x.  4.x is still in LTS until April so I'd like to keep compatibility with that.  The same goes for the use of const/let.  Once 4.x is end of life the code can probably be cleaned up a bit to take advantage of new newer features.

I've just merged the use of `eslint` and Travis to catch these in future.  This branch is will need rebasing as it now conflicts.

**Comment by grantholle** [comment]
So I was going to try and get everything compatible but... 

Should `es6: true` be added to the eslint's `env` property? Embarrassingly I'm not too familiar with eslint 🙈

I'd hate to wait until April but I might. Thoughts?

**Comment by dbuezas** [comment]
I'd love if this could be published

**Comment by alexwhitman** [comment]
@grantholle I've had some time to look at this.  It's close to being mergeable but when running the `'should send clipboard using a callback'` test it throws an `UnhandledPromiseRejectionWarning`.  We shouldn't be getting those when using the callback version.

**Comment by grantholle** [comment]
Ok, that should be working as expected now.

**Comment by alexwhitman** [comment]
Thanks @grantholle.  I've squashed and merged this in https://github.com/alexwhitman/node-pushbullet-api/commit/504da45c26aad20e9f7bd9abb3ff4e3307ce329d

Version 2.1.0 will be released with this.

**Pull request closed without merging**